### PR TITLE
Some cleanup now possible because @jonatanklosko pulled registration

### PR DESCRIPTION
**Pull request by jfly**
events out into a separate join table. This is the start of work for #275.

Also fixed a bug where the validation error on events was not being rendered. See fixed version here:

image

@jonatanklosko, this involved touching your `_associated_events_picker.html.erb`, can you let me know what you think? Thanks!

### CODE REVIEW

File: WcaOnRails/app/views/shared/_associated_events_picker.html.erb [hunk 1]
```diff
@@ -18,9 +18,10 @@ Variables:
      associated_events.find_by_event_id(event.id) || associated_events.build(event_id: event.id)
    end %>
 
-<div class="form-group">
-  <%= label_tag events_association_name, label, class: "associated-events-label" %>
-  <div id="<%= events_association_name %>" class="associated-events">
+<% errors = form_builder.object.errors.messages[events_association_name] || [] %>
+<div class="form-group <%= "has-error" if errors.length > 0 %>">
```
Comment (jonatanklosko) [inline WcaOnRails/app/views/shared/_associated_events_picker.html.erb]: What about
```ruby
  "has-error" unless errors.empty? 
```

File: WcaOnRails/app/views/shared/_associated_events_picker.html.erb [hunk 2]
```diff
@@ -18,9 +18,10 @@ Variables:
...
+  <%= label_tag events_association_name, label, class: "control-label col-sm-2" %>
+  <div id="<%= events_association_name %>" class="col-sm-9">
```
Comment (jonatanklosko) [inline WcaOnRails/app/views/shared/_associated_events_picker.html.erb]: You've added these `col-sm-x` classes directly here but this input isn't always inside a horizontal form. Does it look well on the user profile edit page? Regardless the answer, I consider extending the classes only when it's a part of a horizontal form (as before) to be a bit better.

File: WcaOnRails/app/views/shared/_associated_events_picker.html.erb (hunk 1 above)
Comment (jfly) [inline WcaOnRails/app/views/shared/_associated_events_picker.html.erb]: I'm still not that used to `unless`, but I suppose I should practice it more.

File: WcaOnRails/app/views/shared/_associated_events_picker.html.erb (hunk 2 above)
Comment (jfly) [inline WcaOnRails/app/views/shared/_associated_events_picker.html.erb]: Ah, good catch! I removed these classes while debugging something else, and then was too lazy to add them back. I didn't realize I changed the behavior.

Comment (jfly) [inline WcaOnRails/app/views/shared/_associated_events_picker.html.erb]: I'll change this back to how it was before.

File: WcaOnRails/app/controllers/registrations_controller.rb [hunk 3]
```diff
@@ -39,14 +39,14 @@ def psych_sheet_event
...
-    }.sort_by { |r|
-      has_competed = !!r.world_rank(@event, @preferred_format.sort_by)
-      [ has_competed ? 0 : 1, r.world_rank(@event, @preferred_format.sort_by) || Float::INFINITY, r.world_rank(@event, @preferred_format.sort_by_second) || Float::INFINITY, r.name ]
-    }
+    @registrations = @competition.registrations.
+                                  accepted.
+                                  joins(:registration_events).
+                                  where("registration_events.event_id=?", @event.id).
+                                  sort_by do |r|
+                                    has_competed = !!r.world_rank(@event, @preferred_format.sort_by)
```
Comment (jonatanklosko) [inline WcaOnRails/app/controllers/registrations_controller.rb]: What about adding a method `has_competed_in?(event)` to the user model? Btw. I think that I saw such a line somewhere else, so probably we would be able to reuse it even now =)

Comment (jfly) [inline WcaOnRails/app/controllers/registrations_controller.rb]: I like this idea, but I'm going to wait to do it until #275 is taken care of (at the end of the month). See my comment here: https://github.com/cubing/worldcubeassociation.org/issues/275#issuecomment-217742577

**Comment by jonatanklosko** [comment]
LGTM!

**Pull request merged**